### ISSUE TITLE
Validation error for ColorManager

### ISSUE THREAD

**Issue by mfkeles** [DeepLabCut/napari-deeplabcut#23]
When importing a labeled data folder, I get the following error ("a5" below is a body part name):

```Viewer(axes=Axes(visible=False, labels=True, colored=True, dashed=False, arrows=True), camera=Camera(center=(0.0, 0.0, 0.0), zoom=1.0, angles=(0.0, 0.0, 90.0), perspective=0, interactive=True), cursor=Cursor(position=(1, 1), scaled=True, size=1, style=<CursorStyle.STANDARD: 'standard'>), dims=Dims(ndim=2, ndisplay=2, last_used=0, range=((0, 2, 1), (0, 2, 1)), current_step=(0, 0), order=(0, 1), axis_labels=('0', '1')), grid=GridCanvas(stride=1, shape=(-1, -1), enabled=False), layers=[], scale_bar=ScaleBar(visible=False, colored=False, ticks=True, position=<Position.BOTTOM_RIGHT: 'bottom_right'>, font_size=10, unit=None), text_overlay=TextOverlay(visible=False, color=(0.5, 0.5, 0.5, 1.0), font_size=10, position=<TextOverlayPosition.TOP_LEFT: 'top_left'>, text=''), overlays=Overlays(interaction_box=InteractionBox(points=None, show=False, show_handle=False, show_vertices=False, selection_box_drag=None, selection_box_final=None, transform_start=<napari.utils.transforms.transforms.Affine object at 0x7eff74054160>, transform_drag=<napari.utils.transforms.transforms.Affine object at 0x7eff740541c0>, transform_final=<napari.utils.transforms.transforms.Affine object at 0x7eff74054220>, transform=<napari.utils.transforms.transforms.Affine object at 0x7eff74054280>, allow_new_selection=True, selected_vertex=None)), help='', status='Ready', tooltip=Tooltip(visible=False, text=''), theme='dark', title='napari', mouse_move_callbacks=[<function InteractionBoxMouseBindings.initialize_mouse_events.<locals>.mouse_move at 0x7efe9b37a310>], mouse_drag_callbacks=[<function InteractionBoxMouseBindings.initialize_mouse_events.<locals>.mouse_drag at 0x7efe9b3e6700>], mouse_double_click_callbacks=[], mouse_wheel_callbacks=[<function dims_scroll at 0x7eff76343a60>], _persisted_mouse_event={}, _mouse_drag_gen={}, _mouse_wheel_gen={}, keymap={'Shift': <function InteractionBoxMouseBindings.initialize_key_events.<locals>.hold_to_lock_aspect_ratio at 0x7efe9b3e6c10>, 'Control-Shift-R': <function InteractionBoxMouseBindings._reset_active_layer_affine at 0x7efe9b46f790>, 'Control-Shift-A': <function InteractionBoxMouseBindings._transform_active_layer at 0x7efe9b46f8b0>})
>>> WARNING: Traceback (most recent call last):
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/_qt/qt_viewer.py", line 754, in _qt_open
    self.viewer.open(
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/components/viewer_model.py", line 941, in open
    layers = self._open_or_raise_error(
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/components/viewer_model.py", line 1064, in _open_or_raise_error
    raise MultipleReaderError(
napari.errors.reader_errors.MultipleReaderError: Multiple plugins found capable of reading /home/grover/Desktop/dockerdir/LiquidFoodDLC-Mehmet-2020-03-23/labeled-data/20211118_Fly39_F_B_SD_6D_9am-11182021165425. Select plugin from {'napari-deeplabcut': 'napari DeepLabCut', 'napari': 'napari (npe2)', 'builtins': 'builtins (npe1)'} and pass to reading function e.g. `viewer.open(..., plugin=...)`.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/_qt/qt_viewer.py", line 1106, in dropEvent
    self._qt_open([filename], stack=bool(shift_down))
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/_qt/qt_viewer.py", line 772, in _qt_open
    handle_gui_reading(filenames, self, stack, **kwargs)
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/_qt/dialogs/qt_reader_dialog.py", line 156, in handle_gui_reading
    open_with_dialog_choices(
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/_qt/dialogs/qt_reader_dialog.py", line 249, in open_with_dialog_choices
    qt_viewer.viewer.open(paths, stack=stack, plugin=plugin_name, **kwargs)
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/components/viewer_model.py", line 931, in open
    self._add_layers_with_plugins(
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/components/viewer_model.py", line 1160, in _add_layers_with_plugins
    added.extend(self._add_layer_from_data(*_data))
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/components/viewer_model.py", line 1234, in _add_layer_from_data
    layer = add_method(data, **(meta or {}))
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/components/viewer_model.py", line 4, in add_points
    import itertools
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/layers/points/points.py", line 433, in __init__
    self._face = ColorManager._from_layer_kwargs(
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/layers/utils/color_manager.py", line 577, in _from_layer_kwargs
    return cls(**color_kwargs)
  File "/home/grover/anaconda3/envs/napari-env/lib/python3.8/site-packages/napari/utils/events/evented_model.py", line 197, in __init__
    super().__init__(**kwargs)
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for ColorManager
__root__
  could not convert string to float: 'a5' (type=value_error)
```

**Comment by felixhol** [comment]
same here - I'll start digging where it comes from

How many bodyparts do you have @mfkeles ? I have created a series of identical projects, only varying the number of bodyparts (I'd like to have 36 markers). It seems that somewhere between 14 and 16 bodyparts this problem pops up. Not sure yet why exactly, but it looks like the error appears when the bodyparts are mapped to a colormap.

**Comment by mfkeles** [comment]
sounds about right, was suspecting that. i have 29 body parts.

**Comment by MMathisLab** [comment]
awe, yes it's possible the colormap runs out and does not loop; we can take care of that (@jeylau is on holiday, but should be doable in the next weeks!)

**Comment by jeylau** [comment]
Thanks for documenting the issue @felixhol @mfkeles! I figured out that napari incorrectly guesses the type of a property's variables when there are more than 16 of them, which is now patched with #25.

**Comment by felixhol** [comment]
thanks @jeylau !!!
